include/seqan/graph_types/graph_impl_directed.h

```diff
@@ -88,19 +89,69 @@ class Graph<Directed<TCargo, TSpec> >
             data_allocator(_other.data_allocator)
         {
             SEQAN_CHECKPOINT
-            _copyGraph(_other, *this);
+            _copyGraph(*this, _other);
         }
 
-        Graph const& operator = (Graph const & _other) {
+        Graph const & operator = (Graph const & _other)
+        {
             SEQAN_CHECKPOINT
-            if (this == &_other) return *this;
+            if (this == &_other)
+                return *this;
+
             clear(*this);
             data_allocator = _other.data_allocator;
-            _copyGraph(_other, *this);
+            _copyGraph(*this, _other);
             return *this;
         }
 };
 
+template <typename TCargo>
+class Graph<Directed<TCargo, WithSourceId> >
+{
+    public:
+        typedef typename VertexIdHandler<Graph>::Type TVertexIdManager_;
+        typedef typename EdgeIdHandler<Graph>::Type TEdgeIdManager_;
+        typedef typename EdgeType<Graph>::Type TEdgeStump_;
+        typedef Allocator<SinglePool<sizeof(TEdgeStump_)> > TAllocator_;
+
+        String<TEdgeStump_*> data_vertex;            // Pointers to EdgeStump lists
+        String<TEdgeStump_*> data_vertex_in;         
+        TVertexIdManager_ data_id_managerV;
+        TEdgeIdManager_ data_id_managerE;
+        TAllocator_ data_allocator;
+
+//____________________________________________________________________________
+
+        Graph()
+        {
+            SEQAN_CHECKPOINT
```

Comment:
Please remove all instances of SEQAN_CHECKPOINT (only in the new code). There are not used anymore.